<Type Name="ArithmeticException" FullName="System.ArithmeticException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="92abe27c359490ae14e1421999e10ae2fd43a9a4" /><Meta Name="ms.sourcegitcommit" Value="8ae9a18b248625c8e1851e607c00346a64849587" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="ja-JP" /><Meta Name="ms.lasthandoff" Value="12/17/2019" /><Meta Name="ms.locfileid" Value="75173546" /></Metadata><TypeSignature Language="C#" Value="public class ArithmeticException : SystemException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit ArithmeticException extends System.SystemException" />
  <TypeSignature Language="DocId" Value="T:System.ArithmeticException" />
  <TypeSignature Language="VB.NET" Value="Public Class ArithmeticException&#xA;Inherits SystemException" />
  <TypeSignature Language="C++ CLI" Value="public ref class ArithmeticException : SystemException" />
  <TypeSignature Language="F#" Value="type ArithmeticException = class&#xA;    inherit SystemException" />
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.10.0</AssemblyVersion>
    <AssemblyVersion>4.0.20.0</AssemblyVersion>
    <AssemblyVersion>4.1.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
    <AssemblyVersion>4.2.2.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>1.0.5000.0</AssemblyVersion>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.1.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
    <BaseTypeName FrameworkAlternate="netframework-4.5">System.SystemException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute FrameworkAlternate="netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0;netframework-4.8">
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
    <Attribute FrameworkAlternate="netframework-1.1;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0">
      <AttributeName>System.Serializable</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="7bafa-101">この例外は、キャスト演算または変換演算にエラーがあった場合にスローされます。</span><span class="sxs-lookup"><span data-stu-id="7bafa-101">The exception that is thrown for errors in an arithmetic, casting, or conversion operation.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="7bafa-102"><xref:System.ArithmeticException> は、次の例外の基本クラスです。</span><span class="sxs-lookup"><span data-stu-id="7bafa-102"><xref:System.ArithmeticException> is the base class for the following exceptions:</span></span>  
  
-   <span data-ttu-id="7bafa-103"><xref:System.DivideByZeroException>。除数が0の場合に整数除算でスローされます。</span><span class="sxs-lookup"><span data-stu-id="7bafa-103"><xref:System.DivideByZeroException>, which is thrown in integer division when the divisor is 0.</span></span>  <span data-ttu-id="7bafa-104">たとえば、10を0で除算しようとすると、<xref:System.DivideByZeroException> 例外がスローされます。</span><span class="sxs-lookup"><span data-stu-id="7bafa-104">For example, attempting to divide 10 by 0 throws a <xref:System.DivideByZeroException> exception.</span></span>  
  
-   <span data-ttu-id="7bafa-105"><xref:System.NotFiniteNumberException>。操作がに対して実行されるか、<xref:System.Double.NaN?displayProperty=nameWithType>、<xref:System.Double.NegativeInfinity?displayProperty=nameWithType>、<xref:System.Double.PositiveInfinity?displayProperty=nameWithType>、<xref:System.Single.NaN?displayProperty=nameWithType>、<xref:System.Single.NegativeInfinity?displayProperty=nameWithType>、<xref:System.Single.PositiveInfinity?displayProperty=nameWithType>を返し、使用されているプログラミング言語でこれらの値がサポートされていない場合にスローされます。</span><span class="sxs-lookup"><span data-stu-id="7bafa-105"><xref:System.NotFiniteNumberException>, which is thrown when an operation is performed on or returns <xref:System.Double.NaN?displayProperty=nameWithType>, <xref:System.Double.NegativeInfinity?displayProperty=nameWithType>, <xref:System.Double.PositiveInfinity?displayProperty=nameWithType>, <xref:System.Single.NaN?displayProperty=nameWithType>, <xref:System.Single.NegativeInfinity?displayProperty=nameWithType>, <xref:System.Single.PositiveInfinity?displayProperty=nameWithType>, and the programming language used does not support those values.</span></span>  
  
-   <span data-ttu-id="7bafa-106"><xref:System.OverflowException>。操作の結果が対象のデータ型の範囲外にある場合にスローされます。</span><span class="sxs-lookup"><span data-stu-id="7bafa-106"><xref:System.OverflowException>, which is thrown when the result of an operation is outside the bounds of the target data type.</span></span> <span data-ttu-id="7bafa-107">つまり、数値の `MinValue` プロパティより小さいか、`MaxValue` プロパティよりも大きい値です。</span><span class="sxs-lookup"><span data-stu-id="7bafa-107">That is, it is less than a number's `MinValue` property or greater than its `MaxValue` property.</span></span>  <span data-ttu-id="7bafa-108">たとえば、200 + 200 を <xref:System.Byte> 値に割り当てようとすると、400を超える <xref:System.OverflowException> 例外がスローされます。これは、256よりも <xref:System.Byte> データ型の上限です。</span><span class="sxs-lookup"><span data-stu-id="7bafa-108">For example, attempting to assign 200 + 200 to a <xref:System.Byte> value throws an <xref:System.OverflowException> exception, since 400 greater than 256, the upper bound of the <xref:System.Byte> data type.</span></span>  
  
 <span data-ttu-id="7bafa-109">コードでは、この例外を処理したり、スローしたりしないでください。</span><span class="sxs-lookup"><span data-stu-id="7bafa-109">Your code should not handle or throw this exception.</span></span> <span data-ttu-id="7bafa-110">代わりに、その派生クラスのいずれかを処理またはスローする必要があります。これは、エラーの正確な性質をより正確に示すためです。</span><span class="sxs-lookup"><span data-stu-id="7bafa-110">Instead, you should either handle or throw one of its derived classes, since it more precisely indicates the exact nature of the error.</span></span>  
  
 <span data-ttu-id="7bafa-111"><xref:System.ArithmeticException>インスタンスの初期プロパティ値の一覧については、<xref:System.ArithmeticException>コンストラクターを参照してください。</span><span class="sxs-lookup"><span data-stu-id="7bafa-111">For a list of initial property values for an instance of <xref:System.ArithmeticException>, see the <xref:System.ArithmeticException> constructors.</span></span>  
  
 <span data-ttu-id="7bafa-112"><xref:System.ArithmeticException> では、値0x80070216 を持つ HRESULT COR_E_ARITHMETIC を使用します。</span><span class="sxs-lookup"><span data-stu-id="7bafa-112"><xref:System.ArithmeticException> uses the HRESULT COR_E_ARITHMETIC, which has the value 0x80070216.</span></span>  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Exception" />
    <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="7bafa-113">.NET での例外の処理とスロー</span><span class="sxs-lookup"><span data-stu-id="7bafa-113">Handling and throwing exceptions in .NET</span></span></related>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="7bafa-114"><see cref="T:System.ArithmeticException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="7bafa-114">Initializes a new instance of the <see cref="T:System.ArithmeticException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ArithmeticException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.ArithmeticException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ArithmeticException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
        <AssemblyVersion>4.2.2.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="7bafa-115"><see cref="T:System.ArithmeticException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="7bafa-115">Initializes a new instance of the <see cref="T:System.ArithmeticException" /> class.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="7bafa-116">このコンストラクターは、"算術演算は許可されていません。" などのエラーを説明するシステム提供のメッセージに、新しいインスタンスの <xref:System.Exception.Message%2A> プロパティを初期化します。</span><span class="sxs-lookup"><span data-stu-id="7bafa-116">This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as "The arithmetic operation is not allowed."</span></span> <span data-ttu-id="7bafa-117">このメッセージは、システムの現在のカルチャを考慮して指定します。</span><span class="sxs-lookup"><span data-stu-id="7bafa-117">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="7bafa-118">次の表に、<xref:System.ArithmeticException>のインスタンスのプロパティの初期値を示します。</span><span class="sxs-lookup"><span data-stu-id="7bafa-118">The following table shows initial property values for an instance of <xref:System.ArithmeticException>.</span></span>  
  
|<span data-ttu-id="7bafa-119">プロパティ</span><span class="sxs-lookup"><span data-stu-id="7bafa-119">Property</span></span>|<span data-ttu-id="7bafa-120">値</span><span class="sxs-lookup"><span data-stu-id="7bafa-120">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="7bafa-121">`null` で初期化します。</span><span class="sxs-lookup"><span data-stu-id="7bafa-121">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="7bafa-122">ローカライズされたエラー メッセージの文字列です。</span><span class="sxs-lookup"><span data-stu-id="7bafa-122">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ArithmeticException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.ArithmeticException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ArithmeticException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new ArithmeticException : string -&gt; ArithmeticException" Usage="new System.ArithmeticException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
        <AssemblyVersion>4.2.2.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="7bafa-123">エラーを説明する <see cref="T:System.String" />。</span><span class="sxs-lookup"><span data-stu-id="7bafa-123">A <see cref="T:System.String" /> that describes the error.</span></span></param>
        <summary><span data-ttu-id="7bafa-124">指定したエラー メッセージを使用して、<see cref="T:System.ArithmeticException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="7bafa-124">Initializes a new instance of the <see cref="T:System.ArithmeticException" /> class with a specified error message.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="7bafa-125">`message` パラメーターの内容は、ユーザーが理解できる内容にします。</span><span class="sxs-lookup"><span data-stu-id="7bafa-125">The content of the `message` parameter is intended to be understood by humans.</span></span> <span data-ttu-id="7bafa-126">このコンストラクターの呼び出し元は、この文字列が現在のシステム環境向けにローカライズされていることを確認する必要があります。</span><span class="sxs-lookup"><span data-stu-id="7bafa-126">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span></span>  
  
 <span data-ttu-id="7bafa-127">次の表に、<xref:System.ArithmeticException>のインスタンスのプロパティの初期値を示します。</span><span class="sxs-lookup"><span data-stu-id="7bafa-127">The following table shows initial property values for an instance of <xref:System.ArithmeticException>.</span></span>  
  
|<span data-ttu-id="7bafa-128">プロパティ</span><span class="sxs-lookup"><span data-stu-id="7bafa-128">Property</span></span>|<span data-ttu-id="7bafa-129">値</span><span class="sxs-lookup"><span data-stu-id="7bafa-129">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="7bafa-130">Null 参照 (Visual Basic では`Nothing`)。</span><span class="sxs-lookup"><span data-stu-id="7bafa-130">The null reference (`Nothing` in Visual Basic).</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="7bafa-131">エラー メッセージ文字列です。</span><span class="sxs-lookup"><span data-stu-id="7bafa-131">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected ArithmeticException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.ArithmeticException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; ArithmeticException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new ArithmeticException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; ArithmeticException" Usage="new System.ArithmeticException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
        <AssemblyVersion>4.2.2.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Security.SecuritySafeCritical</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" Index="0" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netframework-1.1;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;netstandard-2.0;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0;netstandard-2.1;netcore-3.1" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" Index="1" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netframework-1.1;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;netstandard-2.0;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0;netstandard-2.1;netcore-3.1" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="7bafa-132">シリアル化されたオブジェクトのデータを保持するオブジェクト。</span><span class="sxs-lookup"><span data-stu-id="7bafa-132">The object that holds the serialized object data.</span></span></param>
        <param name="context"><span data-ttu-id="7bafa-133">転送元または転送先に関するコンテキスト情報です。</span><span class="sxs-lookup"><span data-stu-id="7bafa-133">The contextual information about the source or destination.</span></span></param>
        <summary><span data-ttu-id="7bafa-134">シリアル化されたデータを使用して <see cref="T:System.ArithmeticException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="7bafa-134">Initializes a new instance of the <see cref="T:System.ArithmeticException" /> class with serialized data.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="7bafa-135">このコンストラクターは、シリアル化解除の際に呼び出されて、ストリームで送信された例外オブジェクトを再構成します。</span><span class="sxs-lookup"><span data-stu-id="7bafa-135">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span> <span data-ttu-id="7bafa-136">詳細については、次を参照してください。 [XML および SOAP シリアル化](~/docs/standard/serialization/xml-and-soap-serialization.md)します。</span><span class="sxs-lookup"><span data-stu-id="7bafa-136">For more information, see [XML and SOAP Serialization](~/docs/standard/serialization/xml-and-soap-serialization.md).</span></span>  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/serialization/xml-and-soap-serialization.md"><span data-ttu-id="7bafa-137">XML シリアル化および SOAP シリアル化</span><span class="sxs-lookup"><span data-stu-id="7bafa-137">XML and SOAP Serialization</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ArithmeticException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.ArithmeticException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ArithmeticException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new ArithmeticException : string * Exception -&gt; ArithmeticException" Usage="new System.ArithmeticException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
        <AssemblyVersion>4.2.2.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="7bafa-138">例外の原因を説明するエラー メッセージ。</span><span class="sxs-lookup"><span data-stu-id="7bafa-138">The error message that explains the reason for the exception.</span></span></param>
        <param name="innerException"><span data-ttu-id="7bafa-139">現在の例外の原因になった例外。</span><span class="sxs-lookup"><span data-stu-id="7bafa-139">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="7bafa-140"><paramref name="innerException" /> パラメーターが null 参照でない場合は、内部例外を処理する <see langword="catch" /> ブロックで現在の例外が発生します。</span><span class="sxs-lookup"><span data-stu-id="7bafa-140">If the <paramref name="innerException" /> parameter is not a null reference, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="7bafa-141">指定したエラー メッセージおよびこの例外の原因となった内部例外への参照を使用して、<see cref="T:System.ArithmeticException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="7bafa-141">Initializes a new instance of the <see cref="T:System.ArithmeticException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="7bafa-142">前の例外の直接の結果としてスローされる例外については、<xref:System.Exception.InnerException%2A> プロパティに、前の例外への参照を含める必要があります。</span><span class="sxs-lookup"><span data-stu-id="7bafa-142">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="7bafa-143"><xref:System.Exception.InnerException%2A> プロパティは、コンストラクターに渡されたものと同じ値を返します。内部例外値がコンストラクターに渡されなかった場合、`Nothing` プロパティはnull 参照 (Visual Basic の場合は <xref:System.Exception.InnerException%2A>) を返します。</span><span class="sxs-lookup"><span data-stu-id="7bafa-143">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or a null reference (`Nothing` in Visual Basic) if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="7bafa-144">次の表に、<xref:System.ArithmeticException>のインスタンスのプロパティの初期値を示します。</span><span class="sxs-lookup"><span data-stu-id="7bafa-144">The following table shows initial property values for an instance of <xref:System.ArithmeticException>.</span></span>  
  
|<span data-ttu-id="7bafa-145">プロパティ</span><span class="sxs-lookup"><span data-stu-id="7bafa-145">Property</span></span>|<span data-ttu-id="7bafa-146">値</span><span class="sxs-lookup"><span data-stu-id="7bafa-146">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="7bafa-147">内部例外の参照です。</span><span class="sxs-lookup"><span data-stu-id="7bafa-147">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="7bafa-148">エラー メッセージ文字列です。</span><span class="sxs-lookup"><span data-stu-id="7bafa-148">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="7bafa-149">例外の処理とスロー</span><span class="sxs-lookup"><span data-stu-id="7bafa-149">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
  </Members>
</Type>
